engines/script/sandbox.go

```diff
@@ -0,0 +1,138 @@
+package scriptengine
+
+import (
+	"mime"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/taskcluster/taskcluster-client-go"
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+	"github.com/taskcluster/taskcluster-worker/runtime/atomics"
+)
+
+type sandbox struct {
+	engines.SandboxBase
+	context     *runtime.TaskContext
+	engine      *engine
+	cmd         *exec.Cmd
+	folder      runtime.TemporaryFolder
+	resolve     atomics.Once
+	resultSet   engines.ResultSet
+	resultError error
+	resultAbort error
+	log         *logrus.Entry
+	aborted     atomics.Bool
+	done        chan struct{}
+}
+
+func (s *sandbox) run() {
```

Comment:
General thing...
Well, for all practical intends and purposes WaitForResult should always be called...
But I prefer not to have that call be driving the application... Furthermore, I totally expect that there could be more than one call to WaitForResult.

@walac, as a general rule of thumb, I've come to realize that methods like WaitForResult and Abort aren't good places to do be doing logic. Rather they should just initiate something and then wait for the result to be done.

Note: that execution should start before WaitForResult is called.